framework/include/preference/preference.h

```diff
@@ -0,0 +1,149 @@
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_set_bool(char *key, bool value);
+
+/**
+ * @brief Set string value with key in the preference
+ * @details @b #include <preference/preference.h>
+ * @param[in] key a name of the preference value to set
+ * @param[in] value a string value for the given key
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_set_string(char *key, char *value);
+
+/**
+ * @brief Get int value with key in the preference
+ * @details @b #include <preference/preference.h>
+ * @param[in] key a name of the key to retrieve
+ * @param[out] value an int value for the given key
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_get_int(char *key, int *value);
+
+/**
+ * @brief Get double value with key in the preference
+ * @details @b #include <preference/preference.h>
+ * @param[in] key a name of the key to retrieve
+ * @param[out] value a double value for the given key
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_get_double(char *key, double *value);
+
+/**
+ * @brief Get bool value with key in the preference
+ * @details @b #include <preference/preference.h>
+ * @param[in] key a name of the key to retrieve
+ * @param[out] value a bool value for the given key
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_get_bool(char *key, bool *value);
+
+/**
+ * @brief Get string value with key in the preference
+ * @details @b #include <preference/preference.h>
+ * @param[in] key a name of the key to retrieve
+ * @param[out] value a string value for the given key
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_get_string(char *key, char **value);
+
+/**
+ * @brief Remove the value with the given key from the preference
+ * @details @b #include <preference/preference.h>
+ * @param[in] key a name of the key to remove
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_remove(char *key);
+
+/**
+ * @brief Remove all key-value pairs from the preference
+ * @details @b #include <preference/preference.h>
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_remove_all(void);
+
+/**
+ * @brief Check whether the given key exists in the preference
+ * @details @b #include <preference/preference.h>
+ * @param[in] key a name of the key to check
+ * @param[out] existing true if the key exist, otherwise false
+ * @return On success, OK is returned. On failure, ERROR is returned.
+ * @since TizenRT v2.0 PRE
+ */
+int preference_is_existing(char *key, bool *existing);
```

Comment:
how about returning existing instead of using out param?

Reply:
I think user can't know the exact reason of failure.
For example, whether which file operation is failed or the key doesn't exist